src/main/java/org/javarosa/core/model/SelectChoice.java

```diff
@@ -91,15 +111,33 @@ public String getValue() {
     }
 
     public String getChild(String childName) {
-        if (item != null) {
-            TreeElement child = item.getChild(childName, 0);
-            if (child != null) {
-                return child.getValue().getDisplayText();
-            }
+        if (item == null) {
+            throw new IllegalStateException("Can only get the child of a choice from a select from itemset");
         }
+
+        TreeElement child = item.getChild(childName, 0);
+        if (child != null) {
+            return child.getValue().getDisplayText();
+        }
+
         return null;
     }
 
+    public Map<String, String> getAdditionalChildren() {
+        if (item == null) {
+            throw new IllegalStateException("Can only get the child of a choice from a select from itemset");
+        }
+
+        Map<String, String> children = new LinkedHashMap<>();
```

Comment:
I think some users will want this to be preserve the order the attributes were added in, hence the `LinkedHashMap`. I didn't test for order. Should I?

Reply:
> List<Pair<String, String>>

Ok, excellent.

Reply:
Unless there's a better way to get something like a tuple, I prefer to keep the `LinkedHashMap` and add a test on order.

Reply:
Really? I can't even find a `Pair` type in later Java versions. We could bring the Kotlin standard lib into JavaRosa for this? That'll replace Apache Commons in most places I'd imagine.

Reply:
I was thinking maybe we start introducing Kotlin in the next release and maybe it's even a major version bump? Or do you really think it's something casual we just do whenever?

Reply:
Oh I mean just use the standard library (`org.jetbrains.kotlin:kotlin-stdlib`) as an alternative to Apache Commons. We use it from Java a lot in Collect (before eventually converting).

Reply:
🤦🏻‍♀️